pkg/process/checks/process.go

```diff
@@ -62,6 +62,15 @@ type ProcessCheck struct {
 
 	// Create times by PID used in the network check
 	createTimes atomic.Value
+
+	// sysprobeProcessModuleEnabled tells the process check wheither to use the RemoteSystemProbeUtil to gather privileged proces stats
+	sysprobeProcessModuleEnabled bool
+}
+
+// NotifySysprobeProcessModuleEnabled tells the process check that the sysprobe process module is enabled.
+// This means that the process check should ask the system probe for privileged process stats.
+func (p *ProcessCheck) NotifySysprobeProcessModuleEnabled() {
+	p.sysprobeProcessModuleEnabled = true
```

Comment:
Originally, the `ProcessCheck` would look at `AgentConfig.EnabledChecks` to see if `Process Module` was there to know if it should ask the system probe for privileged process stats. Since we're removing `EnabledChecks`, and deprecating `AgentConfig` there isn't really a good way to let the Process Check know that the check is enabled.
I feel this solution is a bit hacky since we modify the global process check, but I've been going back and forth on whether we should do this, or add a parameter to init for all of the checks.